pyeed/network/network.py

```diff
@@ -235,30 +268,46 @@ def export_cytoscape_graph(self, file_path: str):
             file.write(str(cyt_dict))
         print(f"💾 Network exported to {file_path}")
 
-    def _hide_under_threshold(self, threshold):
+    def hide_threshold(self, threshold, threshold_settings_hide: str = 'UNDER_THRESHOLD'):
         p4c.unhide_all(base_url=self._cytoscape_url)
 
         hide_list = []
 
+
         for u, v, d in self._full_network.edges(data=True):
-            if d["identity"] < threshold:
-                hide_list.append("{} (interacts with) {}".format(u, v))
+            if threshold_settings_hide == 'UNDER_THRESHOLD':
+                if d[self.weight] < threshold:
+                    hide_list.append("{} (interacts with) {}".format(u, v))
+            elif threshold_settings_hide == 'ABOVE_THRESHOLD':
+                if d[self.weight] > threshold:
+                    hide_list.append("{} (interacts with) {}".format(u, v))
 
         p4c.hide_edges(hide_list, base_url=self._cytoscape_url)
 
-    def calculate_degree(self, threshold: float = 0.8):
+    def calculate_degree(self, threshold: float = 0.8, threshold_settings_hide: str = 'UNDER_THRESHOLD'):
         # Calculate degree of nodes with filtering
         degree = {}
         for u, v, d in self._full_network.edges(data=True):
-            if d["identity"] > threshold:
-                if u not in degree:
-                    degree[u] = 1
-                else:
-                    degree[u] += 1
-                if v not in degree:
-                    degree[v] = 1
-                else:
-                    degree[v] += 1
+            if threshold_settings_hide == 'UNDER_THRESHOLD':
+                if d[self.weight] > threshold:
+                    if u not in degree:
+                        degree[u] = 1
+                    else:
+                        degree[u] += 1
+                    if v not in degree:
+                        degree[v] = 1
+                    else:
+                        degree[v] += 1
+            elif threshold_settings_hide == 'ABOVE_THRESHOLD':
+                if d[self.weight] <= threshold:
+                    if u not in degree:
+                        degree[u] = 1
+                    else:
+                        degree[u] += 1
+                    if v not in degree:
+                        degree[v] = 1
+                    else:
+                        degree[v] += 1
 
```

Comment:
As `else:` condition that raises `AttributeError`, to ensure one of the conditions is met.